File: ruby/matrix/matrix.rb
class Matrix
  def initialize(str)
    @str = str
  end

  def rows
    @str
      .split("\n")
      .map { |el| el.split(" ")
             .map { |nested_el| nested_el.to_i }
      }
  end

  def columns
    # count = rows.count - 1
    # column_array = Array.new(count) { Array.new }
    #
    # count.times do |i|
    #   rows.each { |row| column_array[i].push row[i] }
    # end
    #
    # column_array
    rows.transpose
  end
end


matrix = Matrix.new("1 2 3\n4 5 6\n7 8 9\n 8 7 6")
matrix.columns

File: python/matrix/matrix.py
import numpy as np

class Matrix(object):
    def __init__(self, matrix_string):
        self.matrix_string = matrix_string
        split_lines = self.matrix_string.split("\n")
        list_as_strings = [el.split(" ") for el in split_lines]
        self.rows = [[int(el) for el in row] for row in list_as_strings]
        self.columns = np.transpose(self.rows).tolist()

    def row(self, index):
        return self.rows[index - 1]

    def column(self, index):
        return self.columns[index - 1]
